### PR TITLE
NightwatchJS Cucumber Integration

### PR DESCRIPTION
**Pull request by ansel2000**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+## Test infrastructure environments
+
+-   [On-premise](#on-premise)
+-   [Docker](#docker)
+-   [BrowserStack](#browserstack)
+
+## Configuring the maximum parallel test threads for this repository
+
+For all the parallel run configuration profiles, you can configure the maximum parallel test threads by adding the `--parallel` cli switch followed by the number of parallels.
```
Comment (princebaretto99) [inline README.md:57]: Please provide an example full CLI command

File: README.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    -   To run the default test scenario (e.g. End to End Scenario) on your own machine, use the following command:
+
+    ```sh
+    npm run bstack-single
+    ```
+
+    To run a specific test file, use the following command with the additional 'tag-name' argument:
+
+    ```sh
+    npm run bstack --tags @<tag-name>
```
Comment (princebaretto99) [inline README.md:164]: Update this with relevant corrections @ansel2000

File: README.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    ```sh
+    npm run bstack-parallel
+    ```
+
+-   Output
+
+    This run profile executes the entire test suite in parallel on a single BrowserStack browser. Please refer to your [BrowserStack dashboard](https://automate.browserstack.com/) for test results.
+
+    -   Note: By default, this execution would run maximum 10 test threads in parallel on BrowserStack. Refer to the section ["Configuring the maximum parallel test threads for this repository"](#Configuring-the-maximum-parallel-test-threads-for-this-repository) for updating the parallel thread count based on your requirements.
```
Comment (princebaretto99) [inline README.md]: Verify and update @ansel2000

File: README.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+## Additional Resources
+
+-   View your test results on the [BrowserStack Automate dashboard](https://www.browserstack.com/automate)
+-   Documentation for writing [Automate test scripts in Nightwatchjs](https://www.browserstack.com/docs/automate/selenium/getting-started/nodejs/nightwatch)
+-   Customizing your tests capabilities on BrowserStack using our [test capability generator](https://www.browserstack.com/automate/capabilities)
+-   [List of Browsers & mobile devices](https://www.browserstack.com/list-of-browsers-and-platforms?product=automate) for automation testing on BrowserStack
+-   [Using Automate REST API](https://www.browserstack.com/automate/rest-api) to access information about your tests via the command-line interface
+-   Understand how many parallel sessions you need by using our [Parallel Test Calculator](https://www.browserstack.com/automate/parallel-calculator?ref=github)
+-   For testing public web applications behind IP restriction, [Inbound IP Whitelisting](https://www.browserstack.com/local-testing/inbound-ip-whitelisting) can be enabled with the [BrowserStack Enterprise](https://www.browserstack.com/enterprise) offering
```
Comment (princebaretto99) [inline README.md:242]: Add Open Issues @ansel2000

File: src/resources/conf/local.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+const Nightwatch = require("nightwatch");
+const browserstack = require("browserstack-local");
+require("dotenv").config();
+
+let bs_local;
+
+try {
+    const accessKey =
+        process.env.BROWSERSTACK_ACCESS_KEY;
```
Comment (princebaretto99) [inline src/resources/conf/local.js]: Keep an || statement for custom values @ansel2000 . Update the README.md file as well

File: src/resources/conf/nightwatch-browserstack.conf.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        enabled: false,
+        path: 'screens',
+        on_failure: true,
+      },
+
+      desiredCapabilities: {
+        browserName: 'chrome',
+      },
+
+      webdriver: {
```
Comment (princebaretto99) [inline src/resources/conf/nightwatch-browserstack.conf.js]: is this crucial @ansel2000

File: src/test/steps/StepDefinitions.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
+const {
+  Given,
+  Then,
+  When,
+  Before,
+  After
+} = require('@cucumber/cucumber');
+const {
```
Comment (princebaretto99) [inline src/test/steps/StepDefinitions.js]: Remove this @ansel2000

**Review by princebaretto99 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ansel2000) [reply README.md:57]: Done

File: README.md (hunk 3 above)
Comment (ansel2000) [reply README.md]: Made the needed changes

File: README.md (hunk 4 above)
Comment (ansel2000) [reply README.md:242]: Added the Open Issues

File: src/resources/conf/local.js (hunk 5 above)
Comment (ansel2000) [reply src/resources/conf/local.js]: Updated local.js and the README.md

File: src/resources/conf/nightwatch-browserstack.conf.js (hunk 6 above)
Comment (ansel2000) [reply src/resources/conf/nightwatch-browserstack.conf.js]: Done

File: src/test/steps/StepDefinitions.js (hunk 7 above)
Comment (ansel2000) [reply src/test/steps/StepDefinitions.js]: Made the needed changes

**Review by princebaretto99 — APPROVED** [review]

**Review by princebaretto99 — APPROVED** [review]

**Pull request merged**